### PR TITLE
Migration and Health Check for Discord Notifications setup as Slack

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
YES - 201

#### Description
Health check for discord notifications setup as a slack notification. This is a warning message as users will get a better experience setting it up as a Discord Notification. 

@Qstick recommended that this is a healthcheck that we needed on [discord](https://discordapp.com/channels/264387956343570434/269962130982174720/910972822447341568)

#### Todos
- [x] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)

### CODE REVIEW

**Comment by Qstick** [comment]
Option two on this would be to migrate them automatically, what are your thoughts on which is cleaner?

**Comment by RobinDadswell** [comment]
Funnily enough when this was running I thought exactly the same thing. The only issue with migrating it is that users may have it setup for the more simplistic messaging that the slack version gives into discord although that will change when I make my changes to the slack notifications anyway. This warning almost serves as a pre-warning for when we get to that point and could migrate discord notifications to slack notifications at that point instead

**Comment by RobinDadswell** [comment]
Happy to make it a migration in the morning instead though as it's not to dissimilar in terms of work to this and would give users a nicer view on the world 😄

**Comment by RobinDadswell** [comment]
> Option two on this would be to migrate them automatically, what are your thoughts on which is cleaner?

Just had a minor thought about doing it as a migrate, it doesn't stop people following old guides etc. and having it as a slack notification in the future - it may be better to have a health check for the future as well as a migrate, thoughts?

**Comment by Qstick** [comment]
> 

Yea I can buy off on that

File: src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        public SlackUrlCheck(INotificationFactory notificationFactory, ILocalizationService localizationService)
+            : base(localizationService)
+        {
+            _notificationFactory = notificationFactory;
+        }
+
+        public override HealthCheck Check()
+        {
+            if (_notificationFactory.GetAvailableProviders().Where(n => n.Name.Equals("Slack")).Any(s => (s.Definition.Settings as SlackSettings).WebHookUrl.Contains("discord")))
```
Comment (Qstick) [inline src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs:25]: Should this be Where `n.Implementation` equals Slack?

**Review by Qstick — CHANGES_REQUESTED** [review]
Looks good, just the one comment

File: src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs (hunk 1 above)
Comment (RobinDadswell) [reply src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs:25]: Very possibly yes 😂😂

Comment (RobinDadswell) [reply src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs]: Moved it to `n.configcontract.equals("slacksettings")`

File: src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            : base(localizationService)
+        {
+            _notificationFactory = notificationFactory;
+        }
+
+        public override HealthCheck Check()
+        {
+            if (_notificationFactory.GetAvailableProviders().Where(n => n.ConfigContract.Equals("SlackSettings")).Any(s => (s.Definition.Settings as SlackSettings).WebHookUrl.Contains("discord")))
+            {
+                return new HealthCheck(GetType(), HealthCheckResult.Warning, _localizationService.GetLocalizedString("DiscordUrlInSlackNotification"));
```
Comment (bakerboy448) [inline src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs:27]: needs a fragment (and wiki update)

Comment (bakerboy448) [reply src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs:27]: go with `discord-as-slack-notification`

Comment (bakerboy448) [reply src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs:27]: Wiki text sound good?

> - You have Discord configured to use a Slack webhook. This is not advised and the native Discord functionality should be used instead.  If you're receiving this that likely means you just configured this by following an outdated guide. Please advise the guide author to update their steps. If you're receiving this after an update, please visit our support channels as it appears the automatic migration did not complete as expected.

File: src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            : base(localizationService)
+        {
+            _notificationFactory = notificationFactory;
+        }
+
+        public override HealthCheck Check()
+        {
+            if (_notificationFactory.GetAvailableProviders().Where(n => n.ConfigContract.Equals("SlackSettings")).Any(s => (s.Definition.Settings as SlackSettings).WebHookUrl.Contains("discord")))
+            {
+                return new HealthCheck(GetType(), HealthCheckResult.Warning, _localizationService.GetLocalizedString("DiscordUrlInSlackNotification"), "#discord-as-slack-notification");
```
Comment (RobinDadswell) [inline src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs:27]: Fix this to show a list of notifications by name

File: src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs (hunk 2 above)
Comment (RobinDadswell) [reply src/NzbDrone.Core/HealthCheck/Checks/SlackUrlCheck.cs]: `If you're receiving this then it likely means you just configured this by following an outdated guide.`

I would also remove the bit about receiving this after an update as the only way people will get this is if they add it after the migration. The migration covers anything with "discord" in the webhookUrl.

**Comment by Qstick** [comment]
Feel like writing a test on this one?

**Comment by RobinDadswell** [comment]
> Feel like writing a test on this one?

For the health check or the migration?

**Review by Qstick — APPROVED** [review]

**Pull request merged**